[naoqi_apps] add naoqi_tablet node/launch for showImage

kochigami opened the pull request:
I made ros service for using `showImage/hideImage` in ALTabletService. 
In the future, I'd like to use this program and `TakePicture` function in ALPhotoCapture.

At first I tried `ShowImage` by choregraphe, and python sample code in documentation inside Pepper, and I found that in order to use this function in a local PC, 
1. preparing the folder to put pictures in advance 
    under `/home/nao/.local/share/PackageManager/apps/`, and 
2. the tablet connected to wifi
are required.

Then, I confirmed this program 
1. by adding img/html folder under `/home/nao/.local/share/PackageManager/apps/img/html/` inside Pepper, 
2. connecting wifi by using `qicli call ALTabletService.configureWifi security ssid key`, and
3. executing `roslaunch naoqi_apps tablet.launch` and 

```
rosservice call /show_image "file_name:
  data: 'cute_pepper.jpg'" 
```

in a local PC. ROS version is indigo, Naoqi version is 2.4.2.26.
It returned `[INFO] [WallTime: 1456128249.651241] Ok, I'll show you cute_pepper.jpg !` and the picture appeared in the tablet.

What I did in order to put a picture in a robot is as follows:
1. ssh nao@ < pepper ip >
2. cd /home/nao/.local/share/PackageManager/apps/
3. mkdir img
4. cd img & mkdir html
5. sending a picture under /home/nao/.local/share/PackageManager/apps/img/html/

I am very sorry to bother you.
If there is any problem, please let me know.

suryaambrose commented on naoqi_apps/nodes/naoqi_tablet.py:
```diff
@@ -0,0 +1,82 @@
+        
+    def handleShowImageSrv(self, req):
+        try:
+            self.image_path = "http://198.18.0.1/apps/img/" + str(req.file_name.data)
```
Is this IP always the same ?

suryaambrose replied:
And isn't "html" missing ? 
"http://198.18.0.1/apps/img/html" ? (as your image is in /home/nao/.local/share/PackageManager/apps/img/html/)

kochigami commented:
Thank you very much for your review.
I am very sorry for long and complicated explanation.

About IP (198.18.0.1):
----------------------------
I think so, but I don't have much confidence.

I found the sample code under the Aldebaran Documentation Naoqi 2.4.2 -> ALTabletService API -> ShowImage. In the sample code, showImage path is written as follows:
```
# Display a local image located in img folder in the root of the web server
# The ip of the robot from the tablet is 198.18.0.1
tabletService.showImage("http://198.18.0.1/img/help_charger.png")
```
When I executed ```ssh nao@<Pepper IP>``` and ```ifconfig```, 
I also found inet addr of usb0 is 198.18.0.1. 

What I am trying to do here is using ShowImage from a local PC. 
My PC connects to a robot and a robot connects to a tablet (IP is 198.18.0.1), but My PC doesn't connect to a tablet. 
However, after connecting a tablet to wifi, I could execute this program. 

I have to learn why connecting wifi enables this program to work. 
(In addition, I don't know how to confirm wlan IP of a tablet yet.)

About file path:
-------------------
I followed [this tutorial] (http://qiita.com/Atelier-Akihabara/items/87def938b6553ee9c1e9) (I am sorry that this is written in Japanese...) and created Choregraphe app which uses only ShowImage box.
I made html directory to include pictures as the tutorial says, 
but the abusolute file path which is returned by ```_getAbsoluteUrl(url)``` in ShowImage Box doesn't include ```html``` ("http://198.18.0.1/apps/img/<picture name>).

When I put a picture under  ```/home/nao/.local/share/PackageManager/apps/img/html/```, and executed  ``` ShowImage(http://198.18.0.1/apps/img/<picture file>)``` , it succeeded.
However, when I put a picture under  ```/home/nao/.local/share/PackageManager/apps/img/```, and executed  ``` ShowImage(http://198.18.0.1/apps/img/<picture file>)``` , it didn't succeed.

suryaambrose commented:
All right thank you for the explanation :). I am not familiar with the tablet so I wanted to check but if you followed tutorials and if it works, then I suppose it is correct. 

@vrabaud Ok for me, you can merge (I don't have access rights to do it myself)

vrabaud commented:
@suryaambrose , you now are an admin, with great power ...

kochigami commented:
I also added ```ShowWebView``` service.

Like ```ShowImage```,
When I put ```index.html``` file into 
```/home/nao/.local/share/PackageManager/apps/img/html``` and executed
```
rosservice call /show_webview "file_name:
  data: 'index.html'" 
```
, Pepper showed its content.
It can be hidden by using ```rosservice call /hide_image```.


```index.html``` is like this:
```
<html lang="ja">
  <head>
    <meta charset="UTF-8">
    <style type="text/css">
    </style>
  </head>

  <body bgcolor="pink">
    <h1 align="center"><font color ="red" size="7"><b>Pepper's Dairy</b></h1>
    <hr>
    <h3 align="center"><font color ="magenta" size="5">Today, I met NAO.</h3>
    <p align="center"><font color ="magenta" size="5">I'm happy.<br></p>
   </body>
</html>
```

kochigami commented:
I also added set or get file path services.
It uses https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/9 .
```rosservice call  /set_show_image_folder_path``` changes 'This folder path' below :
/home/nao/.local/share/PackageManager/apps/ 'This folder path' (default is ```img```) /html/
That is because other parts are important to execute ```showImage```.

```
rosservice call /set_show_image_folder_path "folder_path:
  data: '(/)aaa/bbb(/)'"  

rosservice call /get_show_image_folder_path 
folder_path: 
  data: /home/nao/.local/share/PackageManager/apps/aaa/bbb/html
```
If there is any problem, please let me know.
I am sorry for making a lot of changes.

kochigami commented:
I also deleted previous ```/show_webview``` service, and added ```/show_app```, ```/show_webview``` services.
If there is any problem, please let me know. I'm sorry for a lot of changes again.

1. ```/show_image``` : The image should be in ```/home/nao/.local/share/PackageManager/apps/ 'Folder path' (default is img) /html/``` and we can change 'Folder path' by using ```/set_show_image_folder_path```.

2. ```/show_app``` : The index.html file of the app should be in ```/home/nao/.local/share/PackageManager/apps/ 'app name' /html/```. 'app name' is a parameter of this service.

3. ```/show_webview``` : The parameter of this service is url to html file. (ex. http: //www.jsk.t.u-tokyo.ac.jp/index-j.html)

4. ```/hide_image``` : This service deletes every image, app, webview and shows Pepper's bubbles.

```
(show_image example)
rosservice call /show_image "file_name:
  data: 'penguin_shitamuki.jpg'" 
status: 
  data: True

(show_webview example)
rosservice call /show_webview "url:
  data: 'http://www.jsk.t.u-tokyo.ac.jp/index-j.html'" 
status: 
  data: True
 rosservice call /show_webview "url:
  data: 'http://www.u-tokyo.ac.jp/index_j.html'" 
status: 
  data: True

(show_app example)
rosservice call /show_app "app_name:
  data: 'img'" 
status: 
  data: True

(hide_image example)
rosservice call /hide_image (Tablet starts to show bubbles.)

(/(s)get_show_image_folder_path example :this changes the path used in /show_image)
rosservice call /get_show_image_folder_path 
folder_path: 
  data: /home/nao/.local/share/PackageManager/apps/img/html/

rosservice call /set_show_image_folder_path "folder_path:
  data: 'aaa/bbb'" 

rosservice call /get_show_image_folder_path 
folder_path: 
  data: /home/nao/.local/share/PackageManager/apps/aaa/bbb/html/



```

kochigami commented:
- added wifi services

    ALTabletService::configureWifi
    ALTabletService::connectWifi
    ALTabletService::disableWifi
    ALTabletService::disconnectWifi
    ALTabletService::enableWifi
    ALTabletService::forgetWifi
    ALTabletService::getWifiStatus

- modified using srv files instead of making original files
- This still requires https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/7

- added error message when a using robot does not have a tablet

I found wifi setting is only required when we want to show a web page.
Without connecting wifi, we can show an image and an app (I just tried showing a html file.)
The comment above is wrong. Sorry about that...

confirmed with:
- [x] Pepper (naoqi 2.5.5) + indigo
- [x] NAO (naoqi 2.1.4) + indigo (shows error)
- [ ] Pepper (naoqi 2.5.5) + kinetic
- [ ] NAO (naoqi 2.1.4) + kinetic

kochigami commented:
This is a new feature, which may cause unexpected trouble when it is merged.
I close this and improve this in my remote.
I'm sorry to leave this for a long time.
